3d_armor/api.lua

```diff
@@ -702,6 +702,7 @@ armor.equip = function(self, player, itemstack)
 		self:run_callbacks("on_equip", player, index, stack)
 		self:set_player_armor(player)
 		self:save_armor_inventory(player)
+		return old_stack or itemstack
```

Comment:
The function description says `Leftover item stack.`. Hence, why is `old_stack` needed here? I would expect it to return the modified `itemstack`.

Reply:
Yes, the `itemstack` should always have only one item, because armor is registered as a tool: https://github.com/minetest-mods/3d_armor/blob/8f775a64f0a29afbcc63cbcd78938c17609ed187/3d_armor/api.lua#L228
I could change the code to not use `take_item`, to make it more obvious.

Reply:
I think I'll leave it, it's safe enough to ignore the possible misuse.